text/stacks/0000-node-on-tiny.md

```diff
@@ -0,0 +1,60 @@
+# Add to Tiny Run Image to Enable Node Apps
+
+## Summary
+
+At the moment, Node.js applications are not compatible with the tiny builder. Therefor the `nodejs` buildpack does not support the tiny stack.
+
+The tiny build image is already capable of building the Node.js application.
+
+The tiny run image is missing `libstdc++6`and `libgcc1`.
+
+## Motivation
+
+To support running Node.js applications with the tiny builder.
+
+While not all Node.js applications could be supported by that, this could be a feasible approach:
+
+- `tiny`: Run Node.js applications **without** any native extenstions
+- `base`: Run Node.js applications **without** many native extensions
+- `full`: Run Node.js applications **with** many native extensions
+
+This should be desirable as some types of Node.js application would benefit of the reduced image size.
+
+```bash
+apps/node-tiny                        latest     7546c8d45780   41 years ago   130MB
+apps/node-base                        latest     022cb9cea147   41 years ago   202MB
+apps/node-full                        latest     1324267830ec   41 years ago   803MB
+```
+
+Additionally image scans would find less CVEs which would make it easier to assess.
+
+| Image | Size Run Image (MB) | Total CVEs | CVEs >= 7 |
+| --- | --- | --- | --- |
+| run-node-tiny | 19.2 |  27 | 14 |
+| paketobuildpacks/run:tiny | 17.4 | 28 | 14 |
+| paketobuildpacks/run:base | 86.7 | 58 | 40 |
+
+## Detailed Explanation
+
+Install the [`libstdc++6`](https://packages.ubuntu.com/bionic/libstdc++6) and [`libgcc1`](https://packages.ubuntu.com/bionic/libgcc1) packages for the Ubuntu based tiny run image. It looks like this will add around 1.8 MB.
+
+The image scan results when writing this pr "only" showed one additional CVE (5.5) which would need to be assessed.
+
+## Rationale and Alternatives
+
+- Node is written in C++, so it needs `libstdc++6`and `libgcc1` at runtime.
+- This would also allow other apps to use the tiny stack, e.g. Rust, C++, SAP JVM
+- This might be solved with the use of `extensions` which is currently discussed
```

Comment:
Not suggesting that I prefer this, but another alternative in addition to using extensions might be language-specific stacks.

In the past we have floated the idea of introducing language-specific builders in order to avoid the layer limit on builder images. This idea is predicated on the assumption that most users know which composite buildpack they should be using and detecting the difference between say a nodejs and a golang application, isn't really a meaningful feature.

If we had language-specific builders it might make sense to create more variations on the the tiny stack that provide the minimal packages typically needed for each language family.

Reply:
Given that `jam create-stack` has now simplified the process of creating and maintaining stacks, and Paketo has new [automation](https://github.com/paketo-buildpacks/github-config/tree/main/stack) to facilitate stack maintenance, this is more possible now than it ever has been. However, as a Builders maintainer, I'm hesitant to add many more builder repositories to the official Paketo offerings. The burden on the Paketo buildpacks Github org's resources is already significant. I imagine @paketo-buildpacks/stacks-maintainers may feel the same about adding Paketo stacks.

While I could potentially see adding stacks to `paketo-community` based on user demand, I'm more inclined to enable users to maintain their own stacks.

Reply:
If we proceed with adding these libraries it should eliminate this concern around builder proliferation.